R/output_format.R

```diff
@@ -597,6 +601,17 @@ create_output_format <- function(name,
   output_format
 }
 
+output_format_string_from_ext <- function(output_file) {
+  default_format <- "html_document"
+  if (is.null(output_file)) return(default_format)
+  switch(xfun::file_ext(output_file),
+    html = "html_document",
+    pdf = "pdf_document",
+    docx = "word_document",
+    "html_document" # always been the default format in R Markdown
```

Comment:
Instead of hard-coding the format:

```suggestion
    default_format  # always been the default format in R Markdown
```